Userland/Services/WindowServer/ScreenLayout.ipp

```diff
@@ -0,0 +1,227 @@
+        *this = {};
+        return false;
+    }
+    return true;
+}
+
+bool ScreenLayout::save_config(Core::ConfigFile& config_file, bool sync) const
+{
+    config_file.write_num_entry("Screens", "DefaultScreen", main_screen_index);
+
+    size_t index = 0;
+    while (index < screens.size()) {
+        auto& screen = screens[index];
+        auto group_name = String::formatted("Screen{}", index);
+        config_file.write_entry(group_name, "Device", screen.device);
+        config_file.write_num_entry(group_name, "Left", screen.location.x());
+        config_file.write_num_entry(group_name, "Top", screen.location.y());
+        config_file.write_num_entry(group_name, "Width", screen.resolution.width());
+        config_file.write_num_entry(group_name, "Height", screen.resolution.height());
+        config_file.write_num_entry(group_name, "ScaleFactor", screen.scale_factor);
+        index++;
+    }
+    // Prune screens no longer in the layout
+    for (;;) {
+        auto group_name = String::formatted("Screen{}", index++);
+        if (!config_file.has_group(group_name))
+            break;
+        config_file.remove_group(group_name);
+    }
+
+    if (sync && !config_file.sync())
+        return false;
+    return true;
+}
+
+bool ScreenLayout::operator!=(const ScreenLayout& other) const
+{
+    if (this == &other)
+        return false;
+    if (main_screen_index != other.main_screen_index)
+        return true;
+    if (screens.size() != other.screens.size())
+        return true;
+    for (size_t i = 0; i < screens.size(); i++) {
+        if (screens[i] != other.screens[i])
+            return true;
+    }
+    return false;
+}
+
+}
+
+namespace IPC {
+
+bool encode(Encoder& encoder, const WindowServer::ScreenLayout::Screen& screen)
+{
+    encoder << screen.device << screen.location << screen.resolution << screen.scale_factor;
+    return true;
+}
+
+bool decode(Decoder& decoder, WindowServer::ScreenLayout::Screen& screen)
+{
+    String device;
+    if (!decoder.decode(device))
+        return false;
+    Gfx::IntPoint location;
+    if (!decoder.decode(location))
+        return false;
+    Gfx::IntSize resolution;
+    if (!decoder.decode(resolution))
+        return false;
+    int scale_factor = 0;
+    if (!decoder.decode(scale_factor))
+        return false;
+    screen = { device, location, resolution, scale_factor };
+    return true;
+}
+
+bool encode(Encoder& encoder, const WindowServer::ScreenLayout& screen_layout)
+{
+    encoder << screen_layout.screens << screen_layout.main_screen_index;
+    return true;
+}
+
+bool decode(Decoder& decoder, WindowServer::ScreenLayout& screen_layout)
+{
+    Vector<WindowServer::ScreenLayout::Screen> screens;
+    if (!decoder.decode(screens))
+        return false;
+    unsigned main_screen_index = 0;
+    if (!decoder.decode(main_screen_index))
+        return false;
+    screen_layout = { move(screens), main_screen_index };
+    return true;
+}
+
+}
```

Comment:
Missing newline at end of file (even if it's not a .cpp file so not checked by the CI :P)